core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -357,8 +354,9 @@ class KafkaApis(val requestChannel: RequestChannel,
         new UpdateMetadataResponse(new UpdateMetadataResponseData().setErrorCode(Errors.STALE_BROKER_EPOCH.code)))
     } else {
       val deletedPartitions = replicaManager.maybeUpdateMetadataCache(correlationId, updateMetadataRequest)
-      if (deletedPartitions.nonEmpty)
-        groupCoordinator.handleDeletedPartitions(deletedPartitions, requestLocal)
+      if (deletedPartitions.nonEmpty) {
+        groupCoordinator.onPartitionsDeleted(deletedPartitions.asJava, requestLocal.bufferSupplier)
```

Comment:
I see in this method we convert back to scala. Is there a way to avoid? 😅

Reply:
Yeah, I was debating this. The truth is that this called only when the leader changes so it is really infrequent. The alternative that I was considering was to use OptionalInt in the old group coordinator as well. I pushed a path to do it. Let me know what you think.